404.md

```diff
@@ -0,0 +1,24 @@
+---
+permalink: /404.html
+layout: error
```

Comment:
I think we should call this `page-not-found` to be a bit clearer and differentiate from other types of errors